### PR TITLE
refactor: label translation

### PR DESCRIPTION
**Pull request by FabianKramm**
**What issue type does this pull request address?** (keep at least one, remove the others) 
/kind enhancement

### CODE REVIEW

File: pkg/util/translate/translate.go [hunk 1]
```diff
@@ -450,3 +519,25 @@ func KindExists(config *rest.Config, groupVersionKind schema.GroupVersionKind) (
...
+		for k, v := range selector.MatchLabels {
+			if out.MatchLabels == nil {
+				out.MatchLabels = map[string]string{}
+			}
+			out.MatchLabels[k] = v
+		}
+		for _, expr := range selector.MatchExpressions {
+			if out.MatchExpressions == nil {
+				out.MatchExpressions = []metav1.LabelSelectorRequirement{}
+			}
```
Comment (facchettos) [inline pkg/util/translate/translate.go:538]: is this needed? append should take care of it I think

File: pkg/util/translate/translate.go [hunk 2]
```diff
@@ -450,3 +519,25 @@ func KindExists(config *rest.Config, groupVersionKind schema.GroupVersionKind) (
...
+func MergeLabelSelectors(elems ...*metav1.LabelSelector) *metav1.LabelSelector {
+	out := &metav1.LabelSelector{}
+	for _, selector := range elems {
+		if selector == nil {
+			continue
+		}
+		for k, v := range selector.MatchLabels {
+			if out.MatchLabels == nil {
+				out.MatchLabels = map[string]string{}
+			}
```
Comment (facchettos) [inline pkg/util/translate/translate.go:532]: I'd probably take this out of the loop and do 
```go
if len(selector.MatchLabels)>0{
    out.MatchLabels = make(map[string]string, len(selector.MatchLabels))
}

File: pkg/util/translate/translate.go (hunk 1 above)
Comment (FabianKramm) [reply pkg/util/translate/translate.go]: Removed it

File: pkg/util/translate/translate.go [hunk 3]
```diff
@@ -73,6 +73,79 @@ func HostMetadata[T client.Object](ctx *synccontext.SyncContext, vObj T, name ty
...
+
+	newLabelSelector := &metav1.LabelSelector{}
+	if labelSelector.MatchLabels != nil {
+		newLabelSelector.MatchLabels = map[string]string{}
+		for k, v := range labelSelector.MatchLabels {
+			newLabelSelector.MatchLabels[labelFunc(ctx, k)] = v
+		}
+	}
+	if len(labelSelector.MatchExpressions) > 0 {
+		newLabelSelector.MatchExpressions = []metav1.LabelSelectorRequirement{}
```
Comment (facchettos) [inline pkg/util/translate/translate.go:136]: this is not needed

Comment (FabianKramm) [reply pkg/util/translate/translate.go]: Yep true, I fixed that

**Pull request merged**